### PR TITLE
Refactor Web3

### PR DESCRIPTION
**Pull request by jjjhill**
Summary of changes:
- Removed current web3 logic
- Implement Web3Modal for selecting provider (have only tested Metamask)
- Changing accounts and logging out is reflected in the app
- Switching chains auto-refreshes the app
- More CSS improvements on the header
- Added `window.` in front of global variables so devs know they're global

### CODE REVIEW

**Comment by ryanprice** [comment]
Is there a way to make the toast that notifies Network changed, get pinned  ?

**Comment by ryanprice** [comment]
When you switch wallets, can we throw a toast up that says "New Wallet Detected"  ?  i feel I am starting to rely on them for communication from the app of state changes I should know about.. even though I am the one that triggered it lol.. if not.. no worries 👍

**Comment by ryanprice** [comment]
Should we reset all states when switching wallets ? I initiated a transaction in 1 wallet, switched wallets, and got a confirmation toast in the next wallet. I think this is more of a coupling thing.. do we want to tightly couple the app state with the wallet ?.. again. very low priority, as I think the user understands.

**Comment by jjjhill** [comment]
> Should we reset all states when switching wallets ? I initiated a transaction in 1 wallet, switched wallets, and got a confirmation toast in the next wallet. I think this is more of a coupling thing.. do we want to tightly couple the app state with the wallet ?.. again. very low priority, as I think the user understands.

the only thing with this is we'd have to use a different toast API. bootstrap-vue doesnt allow us to programatically close toasts (we could probably find a hack tho just ripping it out of the DOM manually) @mchigit thoughts?

**Comment by jjjhill** [comment]
> Is there a way to make the toast that notifies Network changed, get pinned ?

@ryanprice I just changed it so it auto refreshes on network change (do it ourselves instead of relying on window.ethereum to do it, because they're removing that feature soon too)

**Review by ryanprice — APPROVED** [review]
Nice work man.. this runs smooth for me

File: src/components/layout/AppFooter.vue [hunk 1]
```diff
@@ -93,14 +87,19 @@
...
-  computed: mapState({
-    isInjected: state => state.web3.isInjected,
-    network: state => NETWORKS[state.web3.networkId],
-  }),
+  computed: {
+    isConnected() {
+      return this.$store.state.web3.isConnected
+    },
+    network() {
+      const hexString = this.$store.state.web3.chainId
```
Comment (mchigit) [inline src/components/layout/AppFooter.vue:98]: When I'm connected to Rinkeby test network, seems like the `chainId` in web3 is null, which caused the Unidentified Network footer

Comment (mchigit) [reply src/components/layout/AppFooter.vue:98]: This only happens when I login to Metamask with default to Rinkeby network. Works fine if I switch network afterwards

**Review by ryanprice — APPROVED** [review]
Looks good ! , toast dsplays on switching wallets and networks

File: src/components/layout/AppHeader.vue [hunk 2]
```diff
@@ -20,54 +20,49 @@
...
+          Connect To Ethereum
+        </b-button>
+
       </b-navbar-nav>
-  
-      <div class="bonusClass" v-if="coinbase != null && bonusReady == 1  && showSpinner == false" v-on:click="GetBonus">
+    </b-collapse>
+        
+    <div id="bonus-boosters">
+      <div class="bonusClass" v-if="web3isConnected && bonusReady == 1  && showSpinner == false" v-on:click="GetBonus">
```
Comment (mchigit) [inline src/components/layout/AppHeader.vue:53]: 😍  😍  I think this solves https://github.com/ryanprice/Cryptoz.cards-Dapp/issues/13

**Review by mchigit — COMMENTED** [review]
LGTM. Just something I noticed, not sure if they need to be fixed now but just jotting them down. We could open new issues as well:

1. When disconnecting, and then reconnecting with Metamask, the notification shows up multiple times, and the number of time it shows actually increases when I repeatedly disconnect then connect:
image

File: src/components/layout/AppFooter.vue (hunk 1 above)
Comment (mchigit) [reply src/components/layout/AppFooter.vue:98]: nvm saw an issue opened for this

**Review by mchigit — APPROVED** [review]
Thanks for doing this 🙏 🙏

File: src/App.vue [hunk 3]
```diff
@@ -74,131 +125,82 @@ export default {
...
-      
-      var subscription = web3.eth.subscribe('newBlockHeaders', function(error, result){
-        if (!error) {
-          //console.log("New block headers incoming..");
-          //console.log(result);
-          return;
+    getWalletInfo: function() {
+      window.web3.eth.getCoinbase((err, coinbase) => {
+        if (err) {
+          throw new Error(err)
```
[the remark this answers is not in the document]
Comment (jjjhill) [reply src/App.vue]: see change :D @mchigit

**Review by mchigit — APPROVED** [review]

**Pull request merged**